csrc/ir_base_nodes.cpp

```diff
@@ -112,6 +112,17 @@ const std::vector<Expr*>& Val::uses() const {
   return uses_;
 }
 
+bool Val::addUse(Expr* expr) {
+  if (std::find(uses_.begin(), uses_.end(), expr) == uses_.end()) {
+    uses_.push_back(expr);
+    if (vtype_ == ValType::TensorView) {
+      fusion()->invalidateUses();
```

Comment:
I still don't see why this must be called even for tensors. As far as I can see, for tensors, `addUse` is only called from `resetTvUses`, so this has no effect. More generally, why adding a use to a Val can result in an inconsistent use list?

Reply:
You're right. We should only need it when an expression is removed. If that happens it will be via `removeExpr` which already calls `invalidateUses()`, so I believe we can remove this part of `registerExpr` as well: https://github.com/NVIDIA/Fuser/pull/123/files#diff-c43e6052e84882cc0ab02b47194413b830f801f16ff7ae8e664132f6385e2742R571-R573

Reply:
When a TV expression is added to a fusion, if the outputs of the expression have dependency paths to any of fusion outputs, we may need to reset the use lists, so I think it makes sense to invalidate them here.